### PR TITLE
Feat/tree+is cid

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

File: test/resolver.spec.js [hunk 1]
```diff
@@ -77,19 +78,36 @@ describe('IPLD format resolver (local)', () => {
...
         expect(paths).to.eql([
           'name',
+          'nest',
+          'nest/foo',
           'nest/foo/bar',
+          'array',
+          'array/0',
           'array/0/a',
           'array/1',
-          'someLink///0',
```
Comment (nicola) [inline test/resolver.spec.js]: can we still test these edge cases?

Comment (diasdavid) [inline test/resolver.spec.js]: What is the edge case you see? This was a bug

File: test/resolver.spec.js [hunk 2]
```diff
@@ -77,19 +78,36 @@ describe('IPLD format resolver (local)', () => {
 
         expect(paths).to.eql([
           'name',
+          'nest',
```
Comment (dignifiedquire) [inline test/resolver.spec.js]: I'm not a 100% sure on the changes to this test case, why are these the correct ones/what was wrong about the ones before?

Comment (diasdavid) [inline test/resolver.spec.js]: The one that was before, was listing the paths of a Buffer as if it was an array, listing a path per byte of the Buffer.

This new test verifies that the things that are listed, are the ones that are resolvable, given the root of the graph. The best way is just to open the object, since it is small, and just list out all the paths yourself to verify it is correct.

File: test/resolver.spec.js (hunk 1 above)
Comment (nicola) [inline test/resolver.spec.js]: Testing `somelink///0` should still resolve `somelink/0`, no?

Comment (diasdavid) [inline test/resolver.spec.js]: Ah, yes, but this is the output of .tree test, not the resolve :) 

You added a patch to enable those cases :)

Comment (nicola) [inline test/resolver.spec.js]: -_- thank you!

**Comment by kumavis** [comment]
@diasdavid seems like isLink is very generic - does it need to be reimplemented on every resolver?
